### PR TITLE
[BUILD] Add build summary.

### PR DESCRIPTION
**Pull request by michal-raska**

### CODE REVIEW

**Comment by michal-raska** [comment]
closes #993

**Comment by michal-raska** [comment]
Also removes forgotten make target.

File: ci/Jenkinsfile2 [hunk 1]
```diff
@@ -5,12 +5,22 @@
...
+import ai.h2o.ci.BuildResult
 
 def utilsLib = new Utils()
 
+BuildResult result = BuildResult.FAILURE
+
+// initialize build summary
+buildSummary('https://github.com/h2oai/datatable', true)
+// use default StagesSummary implementation
+buildSummary.get().addStagesSummary(this, new StagesSummary())
```
Comment (mmalohlava) [inline ci/Jenkinsfile2]: Minor comment regarding API: 

why do you need to create addStagesSummary explicitly?

better would be:
```
buildSummary.get().createStagesSummary(this)
```

**Comment by mmalohlava** [comment]
Looks great! 2 comments:
  - 2 stages failed but summary view shows all green - we need better handling of interrupt/timeout exception (`org.jenkinsci.plugins.workflow.steps.FlowInterruptedException`)
  - the summary page shows field `Exception` with value `null`

Two improvements for summary view:
  - can you make nodes label clickable to point to actual jenkins slave? (e.g., http://mr-0xc1:8080/computer/mr-dl11/)
  - can you make workspace clickable to point to actual node workspace, or "Allocate node" step in "Pipeline step view" (e.g., http://mr-0xc1:8080/job/h2o-py-data-table/job/PR-1129/4/execution/node/49/)

See below:
image

**Pull request merged**